### PR TITLE
Scorch issue when no root mapping.

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Agent.Worker/Build/TFCommandManager.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public string ResolvePath(string serverPath)
             return localPath?.Trim() ?? string.Empty;
         }
 
-        public async Task ScorchAsync() => await RunCommandAsync(FormatFlags.OmitCollectionUrl, "vc", "scorch", "/diff", "/recursive", SourcesDirectory);
+        public async Task ScorchAsync() => await RunCommandAsync(FormatFlags.OmitCollectionUrl, "vc", "scorch", "/diff", "/recursive", "/unmapped", SourcesDirectory);
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TFCommandManager.cs]: The `/unmapped` switch is simply an enhancement. I hadn't noticed this switch before (or it wasn't there?). It instructs scorch to traverse cloaked directories and traverse subdirectories for single-level maps.

File: src/Agent.Worker/Build/TfsVCSourceProvider.cs [hunk 2]
```diff
@@ -72,6 +72,7 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
             }
 
             // Get the workspaces.
+            executionContext.Output(StringUtil.Loc("QueryingWorkspaceInfo"));
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TfsVCSourceProvider.cs]: Querying can take a while. This is an enhancement that has been in the back of my mind.

File: src/Agent.Worker/Build/TfsVCSourceProvider.cs [hunk 3]
```diff
@@ -153,7 +154,21 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
...
+                            await tf.ScorchAsync();
+                        }
+                        catch (ProcessExitCodeException ex)
+                        {
+                            // Scorch failed.
+                            // Warn, drop the folder, and re-clone.
+                            // TODO: Fix tf.exe scorch and remove this catch handler. This catch handler was
+                            //       added because scorch fails to interpret the workspace when a root mapping
+                            //       does not exist.
+                            executionContext.Warning(ex.Message);
```
Comment (ericsciple) [inline src/Agent.Worker/Build/TfsVCSourceProvider.cs]: The exception message is already localized.

File: src/Microsoft.VisualStudio.Services.Agent/Constants.cs [hunk 4]
```diff
@@ -38,7 +38,7 @@ public enum OSPlatform
 
         public static class Agent
         {
-            public static readonly string Version = "2.103.1";
+            public static readonly string Version = "2.104.0";
```
Comment (bryanmacfarlane) [inline src/Microsoft.VisualStudio.Services.Agent/Constants.cs]: OK.  We'll branch for RC1 off of 2.104.0.  We'll want to take a VSO change tomorrow in releases 103 pointing agent to 104.

**Pull request merged**